std-bits/table/src/main/java/org/enso/table/data/column/storage/numeric/AbstractLongStorage.java

```diff
@@ -18,17 +20,60 @@
 import org.enso.table.data.column.operation.map.numeric.comparisons.LessComparison;
 import org.enso.table.data.column.operation.map.numeric.comparisons.LessOrEqualComparison;
 import org.enso.table.data.column.operation.map.numeric.isin.LongIsInOp;
-import org.enso.table.data.column.storage.*;
+import org.enso.table.data.column.storage.BoolStorage;
+import org.enso.table.data.column.storage.ColumnLongStorage;
+import org.enso.table.data.column.storage.ColumnStorageWithNothingMap;
+import org.enso.table.data.column.storage.Storage;
+import org.enso.table.data.column.storage.ValueIsNothingException;
 import org.enso.table.data.column.storage.type.IntegerType;
 import org.enso.table.data.column.storage.type.StorageType;
+import org.enso.table.data.mask.OrderMask;
+import org.enso.table.data.mask.SliceRange;
+import org.enso.table.problems.BlackholeProblemAggregator;
 import org.graalvm.polyglot.Context;
 
-public abstract class AbstractLongStorage extends NumericStorage<Long>
+public abstract class AbstractLongStorage extends Storage<Long>
     implements ColumnLongStorage, ColumnStorageWithNothingMap {
-  public abstract long getItem(int idx);
-
   private static final MapOperationStorage<Long, AbstractLongStorage> ops = buildOps();
 
+  private final long size;
+  private final IntegerType type;
+  protected BitSet isNothing;
+
+  protected AbstractLongStorage(long size, IntegerType type, BitSet isNothing) {
+    this.size = size;
+    this.type = type;
+    this.isNothing = isNothing;
+  }
+
+  @Override
+  public final long getSize() {
+    return size;
+  }
+
+  @Override
+  public IntegerType getType() {
+    return type;
+  }
+
+  @Override
+  public Long getItemBoxed(long index) {
+    return isNothing(index) ? null : getItemAsLong(index);
+  }
+
+  @Override
+  public boolean isNothing(long idx) {
+    return isNothing.get((int) idx);
+  }
```

Comment:
btw. (this was perhaps not done well before either) Shouldn't maybe `isNothing` also check the bounds? Currently it will reply 'not nothing' for things that are outside the bounds (as `BitSet.get` does not do bounds checks - it just returns `false`)

Reply:
I thought so too but hadn't fixed. Will put in check on limits for it.